### PR TITLE
Fix Attendee and Organizer Parsing and Serialization

### LINKED ISSUE #207: Organizer output is wrong
```
$ pip list | grep ics | awk '{print $2}'
0.6
```

Minimal reproduction

```py
>>> from ics import Event
>>> e = Event()
>>> e.organizer = 'TRUCMACHINCHOUETTECHOSEBROLBIDULEMACHINBAZARTRUCMUCHE'
>>> print(e)
BEGIN:VEVENT
TRUCMACHINCHOUETTECHOSEBROLBIDULEMACHINBAZARTRUCMUCHE
UID:f1d79157-3483-4557-9e48-b5f27e7d7bea@f1d7.org
END:VEVENT
```
should output
```
BEGIN:VEVENT
ORGANIZER;TRUCMACHINCHOUETTECHOSEBROLBIDULEMACHINBAZARTRUCMUCHE
UID:f1d79157-3483-4557-9e48-b5f27e7d7bea@f1d7.org
END:VEVENT
```

### LINKED ISSUE #209: After parsing an iCal file, event.attendees is empty
The ical file came from a Calendar component in a Confluence page - I can probably clean up for public consumption if you need a test case.

Has this problem come up before? I thought it must be me doing something wrong.. at first.

### LINKED ISSUE #218: No way to add `partstat` or `role` to Attendee
I'm attempting to get the GMail RSVP button working using an ICS from this library. They look really good but they're not quite there yet. I'd like to add role and partstat to my Organizer and Attendee because that should be the final piece to getting it to work (I got it working using another library in our older app).

Is there anything I can do here other than overriding these classes with my own implementation?

Cheers!
Gabe

PS. If I find this is a bug with my code I'll close it

### PR DESCRIPTION
**Pull request by N-Coder**
This should fix most of the issues with Attendees and Organizer by using proper Parsers and Serializaers. Note that this somewhat breaks compatibility as `str` objects are no longer allowed for these values. We might add automatic conversion and type-checking later with `attrs`.

Hopefully fixes #207, fixes #209, fixes #218, but still needs tests for this.

### CODE REVIEW

File: ics/event.py [hunk 1]
```diff
@@ -287,51 +261,27 @@ def all_day(self):
         # the event may have an end, also given in 'day' precision
         return self._begin_precision == 'day'
 
-    def make_all_day(self, become_all_day=True):
```
Comment (aureooms) [inline ics/event.py]: Is this based on an unmerged PR?

**Comment by N-Coder** [comment]
Jep, this was unintentionally based on (the single revert-commit from) #231. Should no longer be an issue as that PR is merged now. :wink:

**Comment by C4ptainCrunch** [comment]
https://github.com/C4ptainCrunch/ics.py/pull/232/files#diff-9ee99a66451cc2f8bc37e73337d765e8R200-R204 seems enough to prove that #207 is fixed, i'm writing tests for #209 and #218

**Comment by C4ptainCrunch** [comment]
Adding a test for #209 made me discover a small bug (fixed in 430dcd69) but also a weird exception.

While parsing [tests/fixtures/groupscheduled.ics](https://github.com/C4ptainCrunch/ics.py/blob/master/tests/fixtures/groupscheduled.ics#L25) i get an exception: `ValueError: A ATTENDEE must have at least one CN`.
But when looking at the [RFC](https://tools.ietf.org/html/rfc5545#section-3.8.4.1), page 107 it seems to me that the CN is optional.

```

attendee   = "ATTENDEE" attparam ":" cal-address CRLF

attparam   = *(
          ;
          ; The following are OPTIONAL,
          ; but MUST NOT occur more than once.
          ;
          [...]
          (";" cnparam) / (";" dirparam) /
          [...]
          )
``` 

Is this on purpose ? The test uncovering the problem is in 76637ea

**Comment by N-Coder** [comment]
> Is this on purpose ?

Not really. I must have guessed that it is required because we are using the e-mail address as default if no CN is given. So we should remove the `required=True` and maybe also the (imho unnecessary) redundant default if no CN is given?

**Comment by C4ptainCrunch** [comment]
I just removed the `required=True`, for me it's OK to merge :)

**Comment by mafrosis** [comment]
Great stuff! Will give this a test

**Pull request merged**